domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -37,4 +39,21 @@ public async Task<ActionResult> Create([FromBody] CreateTransferAgreement reques
 
         return Created($"api/transfer-agreements/{result.Id}", result);
     }
+
+    [ProducesResponseType(200)]
+    [HttpGet("api/transfer-agreements")]
+    public async Task<ActionResult<List<TransferAgreementResponse>>> GetBySubjectId()
+    {
+        var subject = User.FindSubjectClaim();
+
+        if(!Guid.TryParse(subject, out var subjectId))
```

Comment:
Are we sure we want to expose the reason why we are having a bad request?

If you dont do the check and you cannot parse, i think we still get a bad request

Reply:
Removed